charts/catalog/templates/controller-manager-deployment.yaml

```diff
@@ -0,0 +1,75 @@
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  name: {{ template "fullname" . }}-controller-manager
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "fullname" . }}-controller-manager
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}-controller-manager
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      containers:
+      - name: controller-manager
+        image: {{ .Values.controllerManager.image }}
+        imagePullPolicy: {{ .Values.controllerManager.imagePullPolicy }}
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+        args:
+        - --port
+        - "8080"
+        - --service-catalog-api-server-url
+        {{- if .Values.apiserver.insecure }}
+        - http://{{ template "fullname" . }}-apiserver
+        {{- else }}
+        - https://{{ template "fullname" . }}-apiserver
+        {{- end }}
+        - -v
+        - "{{ .Values.controllerManager.verbosity }}"
+        ports:
+        - containerPort: 8080
+        volumeMounts:
+        - name: service-catalog-cert
+          mountPath: /etc/service-catalog-ssl
+          readOnly: true
+        readinessProbe:
+          httpGet:
+            port: 8080
+            path: /healthz
+          failureThreshold: 1
+          initialDelaySeconds: 10
```

Comment:
Failing at startup while you wait for dependencies to be running is totally OK and normal.  You just have to converge on a working state.